src/operator/nn/pooling.cc

```diff
@@ -46,15 +46,20 @@ static void PoolingParamParser(nnvm::NodeAttrs *attrs) {
     if (param.stride.ndim() == 0) param.stride = Shape2(1, 1);
     if (param.pad.ndim() == 0) param.pad = Shape2(0, 0);
   } else {
-    CHECK_EQ(param.kernel.ndim(), 3U) << param.kernel.ndim()
-                                       << "D pooling not supported";
+      // ignore kernel size only if global_pool not assigned false
+      if (param.global_pool == false) {
+        CHECK_EQ(param.kernel.ndim(), 3U) << param.kernel.ndim()
+            << "D pooling not supported";
+      }
     if (param.stride.ndim() == 0) param.stride = Shape3(1, 1, 1);
     if (param.pad.ndim() == 0) param.pad = Shape3(0, 0, 0);
   }
-  CHECK_EQ(param.stride.ndim(), param.kernel.ndim())
-      << "stride and kernel should have the same length";
-  CHECK_EQ(param.pad.ndim(), param.kernel.ndim())
-      << "pad and kernel should have the same length";
+  if (param.global_pool == false) {
+    CHECK_EQ(param.stride.ndim(), param.kernel.ndim())
+        << "stride and kernel should have the same length";
+    CHECK_EQ(param.pad.ndim(), param.kernel.ndim())
+        << "pad and kernel should have the same length";
+  }
```

Comment:
Simplify the logic here. If you have used global_pool, there is no need to check the stride, kernel or pad.
```c++
if(! param.global_pool) {
// CHECK kernel, pad, stride
}
```